### PR TITLE
test(profiling): more tests for continuous profiling

### PR DESCRIPTION
**Pull request by armcknight**
Aiming to maximize code coverage of the code that's been added for continuous profiling, alongside the minimal tests added along the way.

for #3555; #skip-changelog

### CODE REVIEW

File: SentryTestUtils/Invocations.swift [hunk 1]
```diff
@@ -42,9 +42,9 @@ public class Invocations<T> {
     }
     
     public func record(_ invocation: T) {
-        queue.async(flags: .barrier) {
+//        queue.async(flags: .barrier) {
```
Comment (philipphofmann) [inline SentryTestUtils/Invocations.swift:45]: `m`: Commented code.

**Review by philipphofmann — APPROVED** [review]
LGTM

**Review by philipphofmann — APPROVED** [review]
Again, LTGM, when the CI is green.

**Comment by armcknight** [comment]
I wound up simplifying the UI tests that integration test launch profiling. Since we now have an exhaustive check on all option combinations in unit tests, we don't need to replicate that again in the UI tests.

**Pull request merged**